### DISCUSSION TITLE
Corrupted Per Scan reports

### DISCUSSION THREAD

**Discussion by TPestaj** [CIRDLES/Squid#762] [Q&A] [answered]
It seems that Issue #757 is not solved.  However, it turns out that it does not have to do with the length of the file path, but rather it has to do with writing to a network location.  ("V:" is one of our network drives.)

![image](https://github.com/CIRDLES/Squid/assets/43234276/e8a5452b-ce2a-44d9-b3a4-e08c077f47e7)

What's more, the behaviour when SQUID tries to create these reports on the network location is inconsistent.  Here is a screenshot of what happens when I click 'Generate all reports' four times for the same squid file without making any changes.  The file sizes are not the same (they get smaller, in fact) and none of these files contain the complete data set.  (Reports were generated using v2.04, but the behaviour is similar for every Squid version I tested.)

![image](https://github.com/CIRDLES/Squid/assets/43234276/c504449a-1258-4dd2-aa28-d038b922b001)

Would you have any thoughts as to what the issue is?  It seems to only affect the reports in the 'Per Scan Reports Bundle'.  Not too sure when the problem started, but I am guessing there was a change in our network configuration/operation. I wonder if other Windows users experience issues in a network environment and have determined what parameter(s) affect Squid behaviour?  Could it be a matter of slow network + Squid write speed?  Unfortunately, we don't have a realistic hope of influencing the network configuration or speed in our case.

I have not been able to come up with a satisfying workaround, but perhaps there could be a way to override default output and specify an alternate (ie local) output location for reports (or just the 'Per Scan Bundle'), maybe specified in a separate configuration file or something?  Then we would move the reports to their correct network location to preserve our data structure (we could write a utility to automate this, I guess).  Any better ideas are welcome.

**Comment by bowring** [comment]
I am going to try exporting a zip of the folder of per scan reports.  Thoughts?

**Reply by TPestaj** [reply]
Worth a try.  If it can export complete reports to our network location then I don't mind if they're zipped.

**Comment by bowring** [comment, marked as answer]
see new v2.0.5- the scans folder is written locally, then zipped, the written to the target folder.  Let me know.

**Comment by NicoleRayner** [comment]
I don't think this is user error here but "Generate all Reports" doesn't seem to do anything anymore.  I can make individual reports as usual (it makes the folder and populates it).  And when I call on just the per spot report bundle no zip appears (not sure where to look for the local version to see if that happened).

**Reply by bowring** [reply]
@NicoleRayner -

I just tested it again on Mac and Win11 - works great. 

Try two things:
1) run squid in verbose mode such as **C:\MYJAVA\jdk8u302-full\bin\java.exe -jar Squid3-2.0.5.jar -verbose** and send me the output in a file
2) copy a .squid file to a local location and try opening it from that location with a local copy of Squid3.

Note: you won't find the temporary versions as they are deleted.

Cheers, @bowring

**Reply by NicoleRayner** [reply]
Works fine locally.  Attached is output from verbose when run on a network file. 
[running squid in verbose.txt](https://github.com/CIRDLES/Squid/files/13561927/running.squid.in.verbose.txt)

**Reply by bowring** [reply]
try using the top of the network drive (shorter path) just to confirm it is the network call and not the path length

**Reply by NicoleRayner** [reply]
Same behaviour

**Reply by bowring** [reply]
Could you please send me all the specs for your network drive so I can see if there is a special condition to check for.  Is Squid3 located on the network drive or just the .squid project?  If it is not, try putting Squid3 on the network drive and running it from there.

I am taking off for a month.  Happy Holidays!

**Comment by TPestaj** [comment]
Could you please clarify what you mean by "send me all the specs for your network drive"?

The way we normally run Squid, the .jar file is located on the same network drive as the data/.squid files (different folder, though).

Enjoy your holidays!
